### PR TITLE
mark FileStream.Lock and Unlock as unsupported on macOS

### PR DESCRIPTION
**Pull request by adamsitnik**
the both [call](https://github.com/dotnet/runtime/blob/911b1d395ef4e59c63048003d676e295414c8d15/src/libraries/System.Private.CoreLib/src/System/IO/FileStream.cs#L259-L287) methods that throw on macOS:

https://github.com/dotnet/runtime/blob/911b1d395ef4e59c63048003d676e295414c8d15/src/libraries/System.Private.CoreLib/src/System/IO/FileStream.Lock.OSX.cs#L8-L16

### CODE REVIEW

File: src/libraries/System.Private.CoreLib/src/System/IO/FileStream.cs [hunk 1]
```diff
@@ -271,6 +273,7 @@ public virtual void Lock(long position, long length)
             LockInternal(position, length);
         }
 
+        [UnsupportedOSPlatform("macos")]
```
Comment (stephentoub) [inline src/libraries/System.Private.CoreLib/src/System/IO/FileStream.cs:276]: Presumably the ref needs to be updated, too?

Comment (stephentoub) [reply src/libraries/System.Private.CoreLib/src/System/IO/FileStream.cs:276]: Also, what about overrides? e.g. IsolatedStorageFileStream.Lock/Unlock?

Comment (adamsitnik) [reply src/libraries/System.Private.CoreLib/src/System/IO/FileStream.cs:276]: @stephentoub excellent points (as usual). I've added missing attributes (ref & overrides) PTAL

Comment (adamsitnik) [reply src/libraries/System.Private.CoreLib/src/System/IO/FileStream.cs:276]: I am sorry, I've missed the VB files... let me fix the build and get back to you

**Comment by adamsitnik** [comment]
The CI failures come from #47374, the PR should be ready for review

File: src/libraries/Microsoft.VisualBasic.Core/src/Microsoft/VisualBasic/FileSystem.vb [hunk 2]
```diff
@@ -1000,6 +1000,7 @@ Namespace Microsoft.VisualBasic
             End Try
         End Sub
 
+        <UnsupportedOSPlatform("macos")> 
         Public Function InputString(ByVal FileNumber As Integer, ByVal CharCount As Integer) As String
```
Comment (carlossanlop) [inline src/libraries/Microsoft.VisualBasic.Core/src/Microsoft/VisualBasic/FileSystem.vb:1004]: I wonder if this API's area owner would prefer to reduce the aggressiveness of marking the whole platform as unsupported, and instead only calling `oFile.Lock()` and `oFile.Unlock()` when the platform is not MacOS.

**Review by carlossanlop — APPROVED** [review]
LGTM given that all the unit tests are failing with unrelated failures.

File: src/libraries/Microsoft.VisualBasic.Core/src/Microsoft/VisualBasic/FileSystem.vb (hunk 2 above)
Comment (buyaa-n) [reply src/libraries/Microsoft.VisualBasic.Core/src/Microsoft/VisualBasic/FileSystem.vb:1004]: As far as I know, mono team has a [story ](https://github.com/dotnet/runtime/issues/45193) for adding more attributes for mac/android/ios, they might want to see this cc @marek-safar

Comment (marek-safar) [reply src/libraries/Microsoft.VisualBasic.Core/src/Microsoft/VisualBasic/FileSystem.vb:1004]: That story is for SDKs but we are going to audit libraries as well during #47910 work

**Review by buyaa-n — APPROVED** [review]
LGTM, thanks

**Pull request merged**